src/EnergyPlus/CMakeLists.txt

```diff
@@ -765,37 +765,63 @@ endif()
 
 add_library( energyplusparser STATIC ${INPUTPARSING_SRC} )
 add_dependencies( energyplusparser GenerateEmbeddedEpJSONSchema )
-target_link_libraries( energyplusparser re2 fmt::fmt)
+target_link_libraries( energyplusparser PUBLIC re2 fmt::fmt)
 if(UNIX AND NOT APPLE)
-  target_link_libraries( energyplusparser dl )
+  target_link_libraries( energyplusparser PUBLIC dl)
 endif()
 if (WIN32)
-  target_link_libraries( energyplusparser Shlwapi )
+  target_link_libraries( energyplusparser PUBLIC Shlwapi )
+endif()
+
+# See https://en.cppreference.com/w/cpp/filesystem#Notes
+if ((CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.1) OR
+    ( ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang") AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)))
+  target_link_libraries( energyplusparser PRIVATE stdc++fs )
 endif()
 
 # first we will create a static library of EnergyPlus
 # this will be linked statically to create the DLL and also the unit tests
 add_library( energypluslib STATIC ${SRC} )
-target_link_libraries( energypluslib fmt::fmt objexx sqlite bcvtb epexpat epfmiimport re2 DElight libkiva btwxt Windows-CalcEngine airflownetworklib ssc )
+target_link_libraries( energypluslib
+  PUBLIC fmt::fmt
+         objexx
+         sqlite
+         bcvtb
+         epexpat
+         epfmiimport
+         re2
+         DElight
+         libkiva
+         btwxt
+         Windows-CalcEngine
+         airflownetworklib
+         ssc
+)
```

Comment:
I do like cleanups in the cmake.  I'd like to know how badly this will conflict with the changes in the work from @lefticus  to modernize our cmake (#8425).  I'm not saying you have to change your approach here, I'm just wanting to prepare myself mentally :laughing:

Reply:
I have seen #8425 and aligned my use of PUBLIC/PRIVATE attributes to match, so it shouldn't be that bad. Plus, I'm touching like 4 or 5 blocks in the CMakeLists.txt, so the merge conflicts should be fine. I'd be happy to fix conflicts on whichever PR is merged second.